frontend/src/features/vessel_sidebar/VesselSidebarHeader/index.tsx

```diff
@@ -0,0 +1,136 @@
+import { useCallback } from 'react'
+import styled from 'styled-components'
+
+import { vesselsAreEquals } from '../../../domain/entities/vessel/vessel'
+import { expandRightMenu } from '../../../domain/shared_slices/Global'
+import { setIsFocusedOnVesselSearch } from '../../../domain/shared_slices/Vessel'
+import { getVesselVoyage } from '../../../domain/use_cases/vessel/getVesselVoyage'
+import { showVessel } from '../../../domain/use_cases/vessel/showVessel'
+import { useAppDispatch } from '../../../hooks/useAppDispatch'
+import { useAppSelector } from '../../../hooks/useAppSelector'
+import { MapButtonStyle } from '../../commonStyles/MapButton.style'
+import { MapComponentStyle } from '../../commonStyles/MapComponent.style'
+import { ReactComponent as SearchIconSVG } from '../../icons/Loupe.svg'
+import { VesselSearch } from '../../VesselSearch'
+import { VesselName } from './VesselName'
+
+export function VesselSidebarHeader() {
+  const dispatch = useAppDispatch()
+
+  const { isFocusedOnVesselSearch, selectedVessel, selectedVesselIdentity, vesselSidebarIsOpen } = useAppSelector(
+    state => state.vessel
+  )
+
+  const { healthcheckTextWarning, previewFilteredVesselsMode, rightMenuIsOpen } = useAppSelector(state => state.global)
+
+  const isVesselNameShown = !isFocusedOnVesselSearch && selectedVesselIdentity
+  const isRightMenuShrinked = vesselSidebarIsOpen && !rightMenuIsOpen
+
+  const onSelectVessel = useCallback(
+    vesselIdentity => {
+      if (!vesselsAreEquals(vesselIdentity, selectedVesselIdentity)) {
+        dispatch(showVessel(vesselIdentity, true, false) as any)
+        dispatch(getVesselVoyage(vesselIdentity, undefined, false) as any)
+      }
+      dispatch(setIsFocusedOnVesselSearch(false))
+    },
+    [dispatch, selectedVesselIdentity]
+  )
+
+  return (
+    <>
+      <VesselNameOrInput
+        data-cy="vessel-name"
+        healthcheckTextWarning={!!healthcheckTextWarning}
+        isHidden={previewFilteredVesselsMode}
+        isRightMenuShrinked={isRightMenuShrinked}
+      >
+        {isVesselNameShown && (
+          <VesselName focusOnVesselSearchInput={() => dispatch(setIsFocusedOnVesselSearch(true))} />
+        )}
+        {!isVesselNameShown && (
+          <VesselSearch
+            extendedWidth={500}
+            hasVesselIdInResults={false}
+            isExtended={isFocusedOnVesselSearch || vesselSidebarIsOpen}
+            isLastSearchedVesselsShowed={isFocusedOnVesselSearch || vesselSidebarIsOpen}
+            onClickOutsideOrEscape={() => dispatch(setIsFocusedOnVesselSearch(false))}
```

Comment:
I know it's cumbersome for small calls like that but unmemoized functions like that will always trigger renderings. Rerendering in general is not a big issue when you don't have performance issues with an app.

But we do have serious performance issues in our case :X. So everything that can be optimized rendering-wise, it's better to take it.